Frontend/android-native/app/src/main/java/in/juspay/mobility/utils/RideRequestUtils.java

```diff
@@ -264,4 +269,74 @@ public void openApplication(Context context){
+            String bundle_version = sharedPref.getString("BUNDLE_VERSION","null");
+            String baseUrl = sharedPref.getString("BASE_URL", "null");
+            String deviceDetails = sharedPref.getString("DEVICE_DETAILS", "null");
+            try
+            {
+                //endPoint for driver status
+                String orderUrl = baseUrl + "/driver/setActivity?active=" + status + "&mode=\"" + mode + "\"";
+                Log.d(LOG_TAG, "orderUrl " + orderUrl);
+                //Http connection to make API call
+                HttpURLConnection connection = (HttpURLConnection) (new URL(orderUrl).openConnection());
+                if (connection instanceof HttpsURLConnection)
+                    ((HttpsURLConnection) connection).setSSLSocketFactory(new TLSSocketFactory());
+                connection.setRequestMethod("POST");
+                connection.setRequestProperty("Content-Type", "application/json");
+                connection.setRequestProperty("x-client-version", versionName);
+                connection.setRequestProperty("token", token);
+                connection.setRequestProperty("x-bundle-version", bundle_version);
+                connection.setRequestProperty("x-device", deviceDetails);
+                connection.setDoOutput(true);
+                connection.connect();
+
+                // validating the response code
+                int respCode = connection.getResponseCode();
+                InputStreamReader respReader;
+                Log.d(LOG_TAG, "respCode "+ respCode);
+
+                if(respCode == 200 && startWidget == true){
+                    if (Settings.canDrawOverlays(context)  && !sharedPref.getString(context.getResources().getString(R.string.REGISTERATION_TOKEN), "null").equals("null") && (sharedPref.getString(context.getResources().getString(R.string.ACTIVITY_STATUS), "null").equals("onPause") || sharedPref.getString(context.getResources().getString(R.string.ACTIVITY_STATUS), "null").equals("onDestroy"))) {
+                        Intent widgetService = new Intent(context, WidgetService.class);
+                        context.startActivity(widgetService);
+                    }
+                }
+
+                if ((respCode < 200 || respCode >= 300) && respCode != 302) {
+                    respReader = new InputStreamReader(connection.getErrorStream());
+                    Log.d(LOG_TAG, "in error "+ respReader);
+                } else {
+                    respReader = new InputStreamReader(connection.getInputStream());
+                    Log.d(LOG_TAG, "in 200 "+ respReader);
+                }
+
+                BufferedReader in = new BufferedReader(respReader);
+                String inputLine;
+                while ((inputLine = in.readLine()) != null) {
+                    result.append(inputLine);
+                }
+                updateStorage("DRIVER_STATUS","__failed", context);
```

Comment:
no need to call this function Just use sharedPref.edit().putString("DRIVER_STATUS","__failed").apply();